purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -930,6 +971,45 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         }
     }
 
+    @JvmSynthetic internal fun updatePendingPurchaseQueue() {
+        if (billingWrapper.isConnected()) {
+            debugLog("[QueryPurchases] Updating pending purchase queue")
+            executorService.execute {
+                val querySUBS = billingWrapper.queryPurchases(BillingClient.SkuType.SUBS)
+                val queryINAPP = billingWrapper.queryPurchases(BillingClient.SkuType.INAPP)
+                val queried = mutableMapOf<String, PurchaseWrapper>()
+                if (querySUBS?.responseCode == BillingClient.BillingResponse.OK && queryINAPP?.responseCode == BillingClient.BillingResponse.OK) {
+                    queried.putAll(querySUBS.purchasesList.map { it.purchaseToken.sha1() to PurchaseWrapper(it, BillingClient.SkuType.SUBS)})
+                    queried.putAll(queryINAPP.purchasesList.map { it.purchaseToken.sha1() to PurchaseWrapper(it, BillingClient.SkuType.INAPP)})
+                    debugLog("[QueryPurchases] Queried tokens: $queried")
+                    synchronized(deviceCache) {
+                        val cachedTokens = deviceCache.getSentTokens().toMutableSet()
+                        debugLog("[QueryPurchases] Already sent tokens: $cachedTokens")
+                        val iterator = cachedTokens.iterator()
+                        while(iterator.hasNext()) {
+                            val token = iterator.next()
+                            if (queried.containsKey(token)) {
+                                debugLog("[QueryPurchases] Token $token still active")
+                                queried.remove(token)
+                            } else {
+                                debugLog("[QueryPurchases] Token $token not active,  will be removed from cache")
+                                iterator.remove()
+                            }
+                        }
+                        deviceCache.setSavedTokens(cachedTokens)
+                    }
+                }
+                postPurchases(
+                    queried.values.toList(),
```

Comment:
So we just always post the queried tokens to the backend? What's the `deviceCache` do then?